mantidproject/mantid
# RROA takes polarization correction parameters from parameters file

**Description of work.**

Added an algorithm that extracts polarization correction parameters from the instrument parameters file and use it to set the correction properties.

**To test:**

Add something like this into an instrument parameter file:
```
  <parameter name="polarization_correction_method" type="string">
    <value val="Fredrikze" />
  </parameter>

  <parameter name="polarization_correction_option" type="string">
    <value val="PA" />
  </parameter>

  <parameter name="efficiency_lambda" type="string">
    <value val="0 3 6 10 15 20" />
    <description is="The wavelengths at which the efficiencies are sampled."/>
  </parameter>

  <parameter name="Pp" type="string">
    <value val="0.991 0.992 0.993 0.994 0.995 0.996 " />
    <description is="The values of the Pp efficiency."/>
  </parameter>

  <parameter name="Ap" type="string">
    <value val="0.981 0.982 0.983 0.984 0.985 0.986" />
    <description is="The values of the Ap efficiency."/>
  </parameter>

  <parameter name="Rho" type="string">
    <value val="0.971 0.972 0.973 0.974 0.975 0.976" />
    <description is="The values of the Rho efficiency."/>
  </parameter>

  <parameter name="Alpha" type="string">
    <value val="0.961 0.962 0.963 0.964 0.965 0.966" />
    <description is="The values of the Alpha efficiency."/>
  </parameter>
```

Run RROA with `PolarizationAnalysis` set to `ParameterFile`. See if it works.

Change the parameters to make them wrong input (eg inconsistent sizes, missing parameters, etc). See the error messages are there and helpful.

Fixes #22500

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
